src/container.ts

```diff
@@ -598,40 +593,17 @@ export function createContainer<T extends object = any, U extends object = any>(
       container,
     }
     if (opts?.esModules) {
-      return new Promise((resolvePromise, reject) => {
-        return (realLoadModules(
-          _loadModulesDeps,
-          globPatterns,
-          opts
-        ) as Promise<LoadModulesResult>)
-          .then(() => {
-            resolvePromise(container)
-          })
-          .catch(reject)
+      return (realLoadModules(_loadModulesDeps, globPatterns, opts) as Promise<
+        LoadModulesResult
+      >).then(() => {
+        return container
```

Comment:
`.then(() => container)` is sufficient 😄

Reply:
Of course it is! 🤦‍♂️ 😂